soc/arm/infineon_xmc/4xxx/soc.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2020 Linumiz
+ * Author: Parthiban Nallathambi <parthiban@linumiz.com>
+ *
+ */
+
+#include <kernel.h>
+#include <init.h>
+#include <soc.h>
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+#define PMU_FLASH_WS		(0x3U)
+
+void z_platform_init(void)
+{
+	u32_t temp;
+
+#if defined(CONFIG_FLOAT)
+	/* Enable FPU */
+	SCB->CPACR |= CPACR_CP10_Msk | CPACR_CP11_Msk;
+#endif
```

Comment:
Do we have to allow FPU coproc access here (i.e. before `prep_c`)? If not, this should be removed since the ARM arch port does this in `prep_c`

https://github.com/zephyrproject-rtos/zephyr/blob/b39423e0e27cc08915b6f9dca1b4b57cff5e7c9e/arch/arm/core/aarch32/prep_c.c#L80-L87

Reply:
@stephanosio Thanks for pointing this. It doesn't needs to be earlier than prep_c, removed FPU handling in soc.